### DISCUSSION TITLE
How to build social on Windows

### DISCUSSION THREAD

**Discussion by seankearon** [squidfunk/mkdocs-material#3074] [Q&A] [answered]
I'm trying to build the social cards on my Windows machine, but I'm getting the following errors:

``` 
mkdocs serve
INFO     -  Building documentation...
no library called "cairo" was found
no library called "libcairo-2" was found
cannot load library 'libcairo.so.2': error 0x7e
cannot load library 'libcairo.2.dylib': error 0x7e
cannot load library 'libcairo-2.dll': error 0x7e
```
I've added the social as follows:

```
plugins:
    - search
    - social
    - minify:
```

These are the `mkdocs` packages that I've got loaded:

```
mkdocs                     1.2.2
mkdocs-macros-plugin       0.6.0
mkdocs-material            7.3.0+insiders.3.1.1
mkdocs-material-extensions 1.0.1
mkdocs-minify-plugin       0.4.1
mkdocs-redirects           1.0.3
```

Any help would be much appreciated! :)

**Comment by squidfunk** [comment, marked as answer]
I'm currently improving the docs, and adding more detailed instructions for building on Windows is definitely on the roadmap. In essence, you need to install `libcairo`. A guide with instructions is here: https://www.cairographics.org/download/

As an alternative, you could build the Insiders Docker image yourself, and use that, as it comes with all necessary dependencies 😊

**Reply by seankearon** [reply]
Perfect, thank you!  So, I installed GTK3 from this [release page](https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases), made sure that was on my path and it builds perfectly now.  

I will look to moving over to using your Docker image soon too. 

I appreciate the super-fast response too! :)

**Comment by seankearon** [comment]
It is building beautifully now, but, sadly, it doesn't render as it should.  I think the reason is that the path separator isn't writing correctly when building on Windows:

```
<meta name=twitter:image content=assets/images/social\pirform\automation.png>
```  

Here's the page:

[https://docs.shineforms.co.uk/pirform/free-trial/](https://docs.shineforms.co.uk/pirform/free-trial/)

I can't find any settings that let me force the path separator, but I can probably tweak it in my build script for now.  This [PR](https://github.com/datarobot/mkdocs-redirects/pull/28) seems to address a similar problem.

Do you think that this is something that could be addressed inside the social plugin?

**Reply by squidfunk** [reply]
This is definitely an error with the social plugin, very likely related to path concatenation. As you may have guessed, I'm not a Windows user, which is why that slipped through. Could you create a new issue for this with all the info needed to reproduce? This is also likely the cause of why Twitter isn't seeing the image.

**Reply by seankearon** [reply]
Done: #3077 

I'll test it out when you have a fix to see if it does solve the problem with the card not rendering.

**Comment by MaximilianKohler** [comment]
I ran into the same problem before, and these two commands seemed to solve it: https://stackoverflow.com/a/74998015

Perhaps they should be added to the "Windows" instructions at https://squidfunk.github.io/mkdocs-material/setup/dependencies/image-processing/#windows

**Reply by kamilkrzyskow** [reply]
~~The issue from the OP was caused by a bug, which was later fixed.~~ 
Edit: My bad there are 2 issues mentioned here, and I wrongly assumed that everything is connected to the bug.

I'm on Windows too and I only used the given instructions in the docs to install the dependencies, so I don't think that adding more commands to the docs is necessary.

However, I'm quite surprised seeing how many different answers that StackOverflow question got. Maybe there is something that I'm missing.

1. Run `pip install pillow cairosvg`
2. Install (using default settings) https://github.com/tschoonj/GTK-for-Windows-Runtime-Environment-Installer/releases 
3. Restart the Terminal window. If you're using an editor with an integrated Terminal (ex. VS Code) restart the editor to make sure it doesn't reuse the old PATH, which was updated in step 2
4. It works

If for some reason it doesn't work check if the PATH Environmental Variable was updated properly.

I would assume people don't restart their Terminal windows properly, hence it still doesn't work after installation. So maybe the guide could explicitly point this out.
